src/app/rpmostree-builtin-status.c

```diff
@@ -654,6 +742,270 @@ status_generic (RPMOSTreeSysroot *sysroot_proxy,
+  g_string_append_printf (summary, "%u %s", n, type);
+}
+
+static int
+compare_sec_advisories (gconstpointer ap,
+                        gconstpointer bp)
+{
+  GVariant *a = *((GVariant**)ap);
+  GVariant *b = *((GVariant**)bp);
+
+  RpmOstreeAdvisorySeverity asev;
+  g_variant_get_child (a, 2, "u", &asev);
+
+  RpmOstreeAdvisorySeverity bsev;
+  g_variant_get_child (b, 2, "u", &bsev);
+
+  if (asev != bsev)
+    return asev - bsev;
+
+  const char *aid;
+  g_variant_get_child (a, 0, "&s", &aid);
+
+  const char *bid;
+  g_variant_get_child (b, 0, "&s", &bid);
+
+  return strcmp (aid, bid);
+}
+
+static const char*
+severity_to_str (RpmOstreeAdvisorySeverity severity)
+{
+  switch (severity)
+    {
+    case RPM_OSTREE_ADVISORY_SEVERITY_NONE:
+      return "";
+    case RPM_OSTREE_ADVISORY_SEVERITY_LOW:
+      return "Low";
+    case RPM_OSTREE_ADVISORY_SEVERITY_MODERATE:
+      return "Moderate";
+    case RPM_OSTREE_ADVISORY_SEVERITY_IMPORTANT:
+      return "Important";
+    case RPM_OSTREE_ADVISORY_SEVERITY_CRITICAL:
+      return "Critical";
+    default:
+        return "";
+    }
+}
+
+static void
+print_advisories (GVariant *advisories,
+                  guint     max_key_len)
+{
+  const guint n = g_variant_n_children (advisories);
+  /* counters for none, low, moderate, important, critical advisories */
+  guint n_sev[5] = {0,};
+
+  /* we only display security advisories for now */
+  g_autoptr(GPtrArray) sec_advisories =
+    g_ptr_array_new_with_free_func ((GDestroyNotify)g_variant_unref);
+
+  guint max_id_len = 0;
+
+  GVariantIter iter;
+  g_variant_iter_init (&iter, advisories);
+  while (TRUE)
+    {
+      g_autoptr(GVariant) child = g_variant_iter_next_value (&iter);
+      if (!child)
+        break;
+
+      DnfAdvisoryKind kind;
+      g_variant_get_child (child, 1, "u", &kind);
+
+      /* we only display security advisories for now */
+      if (kind != DNF_ADVISORY_KIND_SECURITY)
+        continue;
+
+      const char *id;
+      g_variant_get_child (child, 0, "&s", &id);
+      max_id_len = MAX (max_id_len, strlen (id));
+
+      RpmOstreeAdvisorySeverity severity;
+      g_variant_get_child (child, 2, "u", &severity);
+      if (RPM_OSTREE_ADVISORY_SEVERITY_NONE < severity &&
+          severity <= RPM_OSTREE_ADVISORY_SEVERITY_CRITICAL)
+        n_sev[severity]++;
+      else
+        n_sev[0]++;
+
+      g_ptr_array_add (sec_advisories, g_variant_ref (child));
+    }
+
+  if (sec_advisories->len == 0)
+    return;
+
+  /* Fedora right now doesn't put severity levels in updateinfo.xml -- print a nicer version
+   * for that case: https://github.com/fedora-infra/bodhi/pull/2099 */
+  const gboolean no_severities = (n_sev[0] == n);
+
+  g_print ("%s%s", red_prefix, bold_prefix);
+  print_kv_no_newline ("SecInfo", max_key_len, "");
+  g_print ("%s%s", bold_suffix, red_suffix);
+
+  if (!opt_verbose)
+    {
+      if (no_severities)
+        g_print ("%u advisor%s\n", n, _N ("y", "ies", n));
```

Comment:
In general I feel like we can avoid these linguistic gymnastics in more cases by doing key/value pairs:
`Security advisories: $n`.  That seems OK even if `$n = 1`.